src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs

```diff
@@ -750,6 +750,250 @@ class C1<T>
+
+            var ptrOriginal = (FunctionPointerTypeSymbol)mOriginal.ReturnType;
+            var ptrRetargeted = (FunctionPointerTypeSymbol)mRetargeted.ReturnType;
+
+            FunctionPointerUtilities.CommonVerifyFunctionPointer(ptrOriginal);
+            FunctionPointerUtilities.CommonVerifyFunctionPointer(ptrRetargeted);
+
+            assert(returnConsistent.typeConsistent,
+                   ptrOriginal.Signature.ReturnType.ContainingAssembly,
+                   ptrRetargeted.Signature.ReturnType.ContainingAssembly);
+            assert(returnConsistent.refModConsistent,
+                   getModifierAssemblySymbol(ptrOriginal.Signature.RefCustomModifiers),
+                   getModifierAssemblySymbol(ptrRetargeted.Signature.RefCustomModifiers));
+            assert(returnConsistent.typeModConsistent,
+                   getModifierAssemblySymbol(ptrOriginal.Signature.ReturnTypeWithAnnotations.CustomModifiers),
+                   getModifierAssemblySymbol(ptrRetargeted.Signature.ReturnTypeWithAnnotations.CustomModifiers));
+
+            Assert.Equal(2, ptrOriginal.Signature.ParameterCount);
+            Assert.Equal(2, ptrRetargeted.Signature.ParameterCount);
+
+            var param1Original = ptrOriginal.Signature.Parameters[0];
+            var param2Original = ptrOriginal.Signature.Parameters[1];
+            var param1Retargeted = ptrRetargeted.Signature.Parameters[0];
+            var param2Retargeted = ptrRetargeted.Signature.Parameters[1];
+
+            assert(param1Consistent.typeConsistent,
+                   param1Original.Type.ContainingAssembly,
+                   param1Retargeted.Type.ContainingAssembly);
+            assert(param1Consistent.refModConsistent,
+                   getModifierAssemblySymbol(param1Original.RefCustomModifiers),
+                   getModifierAssemblySymbol(param1Retargeted.RefCustomModifiers));
+            assert(param1Consistent.typeModConsistent,
+                   getModifierAssemblySymbol(param1Original.TypeWithAnnotations.CustomModifiers),
+                   getModifierAssemblySymbol(param1Retargeted.TypeWithAnnotations.CustomModifiers));
+
+            assert(param2Consistent.typeConsistent,
+                   param2Original.Type.ContainingAssembly,
+                   param2Retargeted.Type.ContainingAssembly);
+            assert(param2Consistent.refModConsistent,
+                   getModifierAssemblySymbol(param2Original.RefCustomModifiers),
+                   getModifierAssemblySymbol(param2Retargeted.RefCustomModifiers));
+            assert(param2Consistent.typeModConsistent,
+                   getModifierAssemblySymbol(param2Original.TypeWithAnnotations.CustomModifiers),
+                   getModifierAssemblySymbol(param2Retargeted.TypeWithAnnotations.CustomModifiers));
+
+            static MethodSymbol getMethodSymbol(CSharpCompilation compilation)
```

Comment:
>static [](start = 12, length = 6)

It feels like we are lacking an assert that the function pointer type stayed the same if all types involved expected to be "consistent".